packages/hub/src/lib/commit.ts

```diff
@@ -459,6 +464,35 @@ export async function* commitIter(params: CommitParams): AsyncGenerator<CommitPr
 
 		yield { event: "phase", phase: "committing" };
 
+		const allOpsToCommit = [...deleteOps, ...fileOpsToCommit];
+		if (allOpsToCommit.length === 0) {
+			const res: Response = await (params.fetch ?? fetch)(
+				`${params?.hubUrl || HUB_URL}/api/${repoId.type}s/${repoId.name}`,
```

Comment:
there doesn't seem to be any method already to call this endpoint, correct?

Reply:
> I think we should include the sha directly in the preupload response

That would simplifies things indeed!

Reply:
see 325d9170d7a25013a485c8103053c9f03eed08d2